File: java93/javascript01/step11/test02_2.js

/* 데이터 조회 명령 : insert
  insert into 테이블명(컬럼명, 컬럼명)
  values(값, 값)
*/
const mysql = require('mysql')
const connection = mysql.createConnection({
  database: 'webappdb',
  user: 'webapp',
  password: '1111'
}) //
console.log('MySQL 서버에 연결할 도구 생성완료')

connection.connect()
console.log('MySQL 서버와 연결됨!')



// query(SQL 명령, 결과가 도착했을 때 호출될 함수)
connection.query('insert into memb(name, email, tel, pwd)'+
              ' values("오호라","ohora@test.com", "111-1111", password("1111"))', function(error, results){

  console.log('결과를 가져왔음!')
  if (error) throw error;
  console.log(results)
})

connection.end()
console.log('MySQL 서버와 연결 끊기가 예약되었음!')

File: java01/src/step20/Test02_2.java
/* JDBC 프로그래밍: Statement 객체 - INSERT 실행
 * => Statement.executeUpdate()
 *    - INSERT, UPDATE, DELETE 등 DML, DDL 을 실행한다.
 *      리턴 값은 명령어의 영향을 받은 레코드 갯수이다.
 *      예) INSERT -> 등록된 레코드 개수,
 *          UPDATE -> 변경된 레코드 개수,
 *          DELETE -> 삭제된 레코드 개수 
 */
package step20;

import java.sql.DriverManager;

public class Test02_2 {
  public static void main(String[] args) throws Exception{
    com.mysql.jdbc.Driver mysqlDriver = new com.mysql.jdbc.Driver();
    java.sql.Connection con = DriverManager.getConnection(
        "jdbc:mysql://localhost:3306/webappdb", 
        "webapp",
        "1111"); 
    
    java.sql.Statement stmt = con.createStatement();
    System.out.println("DBMS와 연결되었습니다.");
    
    int count = stmt.executeUpdate("insert into memb(name,tel,email,pwd)" 
        + "values('홍길동', '1212-1313', 'hong101@test.com', password('1111'))");
    
    stmt.close();
    con.close();
  }
}
